### ISSUE TITLE
Force code on one line if it is possible

### ISSUE THREAD

**Issue by vdhpieter** [rubocop/rubocop#7977]
## Is your feature request related to a problem? Please describe.

I would like a cop that forces ruby code as much as possible on 1 line. I am integrating RuboCop way more strictly in a long-living project. It's going pretty well but the autocorrection sometimes adds more lines than needed IMO. To go even further I would like it if I can enforce code to be on 1 line as much as possible (until the max linewidth is reached). What happens now, for instance, is this:
```ruby
method_call(other_method_call(foo: "this is a verry loooooooong string", bar: 1234), foobar: 123)
```
is corrected to:
```ruby
method_call(
  other_method_call(
    foo: "this is a verry loooooooong string",
    bar: 1234
  ),
  foobar: 123
)
```
And I would like:
```ruby
method_call(
  other_method_call(foo: "this is a verry loooooooong string", bar: 1234),
  foobar: 123
)
```
both are accepted with my current config

## Describe the solution you'd like

A cop with this behavior

```ruby
# bad
{
  foo: 1,
  bar: 2
}

#bad
foo(
  123,
  'bar'
)

# good
{ foo: 1, bar: 2 }

# good
foo(123, 'bar')
```

## Describe alternatives you've considered

I dug pretty deep in RuboCop docs but could not find any thing. Prettier for ruby does do it like this way but I would like to use 1 tool instead of the combination of a lot.

If possible I would be open to help write this. But I never did anything like linting/formatting so I don't think it will be an easy task
